File: Baekjoon_Online_Judge/10_brute_force/07_boj_2839.js
const fs = require("fs");
const input = parseInt(fs.readFileSync("/dev/stdin").toString().trim());

const minCount = [];
for(let five = 0; five<=parseInt(input / 5); five++){
	if((input - five * 5) % 3 !== 0){
		continue;
	}
	minCount.push((input - five * 5) / 3 + five);
}

console.log(minCount.length > 0 ? Math.min(...minCount) : -1);

File: Baekjoon_Online_Judge/10_brute_force/07_boj_2839.cpp
#include <iostream>

int main() {
	int n, min;
	scanf("%d", &n);
	min = n;
	// 5x+3y = n
	for(int i = 0; i<n/5+1; i++){
		int r = n-(5*i);
		if(r%3 == 0 and min > i+r/3)
			min = i+r/3;
	}
	if(min == n)
		min = -1;
	printf("%d", min);
	return 0;
}